.github/workflows/CI_on_push.yml

```diff
@@ -0,0 +1,41 @@
+
+name: CI on push to main or issue-2 branch
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+      - issue-2-Setup_CI/CD_for_reference-models
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: olafurpg/setup-scala@v10
+
+      - name: Install clang and runtime dependencies
+        run: |
+          sudo apt install clang libunwind-dev
+          sudo apt install libgc-dev libre2-dev
```

Comment:
You can combine these two lines like this:
```
sudo apt install clang libunwind-dev libgc-dev libre2-dev
```

Reply:
Done!